plugins/datetime/msd-datetime-mechanism.c

```diff
@@ -113,334 +156,327 @@ msd_datetime_mechanism_error_get_type (void)
+                                     "Expected UTC or LOCAL at line 3 of /etc/adjtime; found '%s'",
+                                     lines[2]);
+                g_dbus_method_invocation_return_gerror (invocation, error);
                 g_error_free (error);
-                g_object_unref (result);
-
+                g_strfreev (lines);
                 return FALSE;
         }
+        g_strfreev (lines);
+        mate_settings_date_time_mechanism_complete_get_hardware_clock_using_utc (object, invocation, is_utc);
+        return TRUE;
+}
 
-        g_object_unref (result);
+static gboolean msd_datetime_mechanism_get_timezone_handler (MateSettingsDateTimeMechanism *object,
+                                                             GDBusMethodInvocation *invocation,
+                                                             gpointer         user_data)
+{
+        gchar *timezone;
+
+        reset_killtimer ();
+
+        timezone = system_timezone_find ();
+
+        mate_settings_date_time_mechanism_complete_get_timezone (object, invocation, timezone);
 
         return TRUE;
 }
 
-
-static gboolean
-_set_time (MsdDatetimeMechanism  *mechanism,
-           const struct timeval  *tv,
-           DBusGMethodInvocation *context)
+static gboolean msd_datetime_mechanism_set_hardware_clock_using_utc_handler (
+        MateSettingsDateTimeMechanism *object,
+        GDBusMethodInvocation *invocation,
+        gboolean using_utc,
+        gpointer user_data)
 {
-        GError *error;
-
-        if (!_check_polkit_for_action (mechanism, context, "org.mate.settingsdaemon.datetimemechanism.settime"))
-                return FALSE;
+        GError *error = NULL;
+        MsdDatetimeMechanism *mechanism;
 
-        if (settimeofday (tv, NULL) != 0) {
-                error = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
-                                     MSD_DATETIME_MECHANISM_ERROR_GENERAL,
-                                     "Error calling settimeofday({%ld,%ld}): %s",
-                                     (gint64) tv->tv_sec, (gint64) tv->tv_usec,
-                                     strerror (errno));
-                dbus_g_method_return_error (context, error);
+        mechanism = MSD_DATETIME_MECHANISM (user_data);
+        if (!_check_polkit_for_action (mechanism,
+                                       "org.mate.settingsdaemon.datetimemechanism.configurehwclock",
+                                       g_dbus_method_invocation_get_sender (invocation),
+                                       &error)) {
+                g_dbus_method_invocation_return_gerror (invocation, error);
                 g_error_free (error);
                 return FALSE;
         }
 
         if (g_file_test ("/sbin/hwclock",
-                         G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR | G_FILE_TEST_IS_EXECUTABLE)) {
+                                G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR | G_FILE_TEST_IS_EXECUTABLE)) {
```

Comment:
Indent

Reply:
If you do not make any changes to the code, please do not format it, as you will increase the number of lines to review.